pkg/ddevapp/config.go

```diff
@@ -212,22 +284,40 @@ func (c *Config) Hostname() string {
 
 // WriteDockerComposeConfig writes a docker-compose.yaml to the app configuration directory.
 func (c *Config) WriteDockerComposeConfig() error {
-	log.WithFields(log.Fields{
-		"Location": c.DockerComposeYAMLPath(),
-	}).Debug("Writing docker-compose.yaml")
+	var err error
 
-	f, err := os.Create(c.DockerComposeYAMLPath())
-	if err != nil {
-		return err
-	}
-	defer util.CheckClose(f)
+	if !fileutil.FileExists(c.GetPath("provider.env")) {
+		log.WithFields(log.Fields{
+			"Location": c.GetPath("provider.env"),
+		}).Debug("Writing blank provider.env")
 
-	rendered, err := c.RenderComposeYAML()
-	if err != nil {
-		return err
+		// Create an empty provider.env file to make docker-compose happy.
+		_, err := os.Create(c.GetPath("provider.env"))
```

Comment:
You will get an empty `provider.env` file regardless of whether you are using pantheon or not.  Otherwise, docker-compose will throw an error when using the `env_file` directive. An empty provider.env is now created alongside the docker-compose.yaml, so the user can commit it or ignore it as they choose.

I don't love this, but we need a way to conditionally include environment vars. Open to other suggestions on how to achieve it.